### PR TITLE
Add unauthorized middleware

### PR DESCRIPTION
**Pull request by elazzabi**
This displays an Unauthorized error and quits when 401 is received from the API

### How to test:
0. Get this branch, `npm i` then `npm link`
1. Be sure you're logged in to production
2. Run `API_HOST="http://localhost:4000" vip app list`
3. You should get: `Error: You are unauthorized to perform this request`

### CODE REVIEW

File: src/lib/api.js [hunk 1]
```diff
@@ -25,8 +27,17 @@ export default async function API(): Promise<ApolloClient> {
 		headers.Authorization = `Bearer ${ token.raw }`;
 	}
 
+	const unauthorizedLink = onError( ( { networkError } ) => {
+		if ( networkError.statusCode === 401 ) {
+			console.error( chalk.red( 'Error:' ), 'You are unauthorized to perform this request' );
```
Comment (nickdaugherty) [inline src/lib/api.js]: Discussed on Slack - let's make this message a little more verbose by including something that says "Please log in again" with details on how to logout `vip logout`.

File: src/lib/api.js [hunk 2]
```diff
@@ -25,8 +27,17 @@ export default async function API(): Promise<ApolloClient> {
 		headers.Authorization = `Bearer ${ token.raw }`;
 	}
 
+	const unauthorizedLink = onError( ( { networkError } ) => {
+		if ( networkError.statusCode === 401 ) {
+			console.error( chalk.red( 'Unauthorized:' ), 'You are unauthorized to perform this request, please use vip logout.' );
```
Comment (nickdaugherty) [inline src/lib/api.js]:
```suggestion
			console.error( chalk.red( 'Unauthorized:' ), 'You are unauthorized to perform this request, please logout with `vip logout` then try again.' );
```

**Review by nickdaugherty — DISMISSED** [review]
Works great! 🚢

**Comment by elazzabi** [comment]
Windows tests are passing, just timing out. Merging this

**Pull request merged**